packages/block-editor/src/components/link-control/input-search.js

```diff
@@ -0,0 +1,69 @@
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import {
+	IconButton,
+} from '@wordpress/components';
+import {
+	LEFT,
+	RIGHT,
+	UP,
+	DOWN,
+	BACKSPACE,
+	ENTER,
+} from '@wordpress/keycodes';
+
+/**
+ * Internal dependencies
+ */
+import {
+	URLInput,
+} from '../';
+
+const LinkControlInputSearch = ( { value, onChange, onSelect, renderSuggestions, fetchSuggestions, onReset } ) => {
+	const stopPropagation = ( event ) => {
+		event.stopPropagation();
+	};
+
+	const stopPropagationRelevantKeys = ( event ) => {
+		if ( [ LEFT, DOWN, RIGHT, UP, BACKSPACE, ENTER ].indexOf( event.keyCode ) > -1 ) {
+			// Stop the key event from propagating up to ObserveTyping.startTypingInTextField.
+			stopPropagation( event );
+		}
+	};
+
+	return (
+		<form>
+			<URLInput
+				className="block-editor-link-control__search-input"
+				value={ value }
+				onChange={ onChange }
+				onKeyDown={ ( event, suggestion ) => {
+					stopPropagationRelevantKeys( event );
+					if ( event.keyCode === ENTER ) {
+						onSelect( event, suggestion );
+					}
+				} }
+				onKeyPress={ stopPropagation }
```

Comment:
is `<URLInput />` handling the `onKeyPress` event? [It doesn't seem to do](https://github.com/WordPress/gutenberg/blob/master/packages/block-editor/src/components/url-input/index.js#L24-L39).
I doubt because it already tries to bind the function in the current implementation of `<NavigationMenuItem />`

Reply:
https://github.com/WordPress/gutenberg/pull/17846/commits/19d5e645b7f742100e238866b033cf0c3113ef14 handles the `onKeyPress` event of the `<URLInput />` component.